### PR TITLE
add integration test for deployment slot

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

File: azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java [hunk 1]
```diff
@@ -9,26 +9,78 @@
...
+                formatCommand(azAppServicePlanCreateTemplate, new String[]{resourceGroupName, servicePlanName}),
+                formatCommand(azWebAppCreateTemplate, new String[]{
+                        resourceGroupName, webAppName, servicePlanName, runtime})
+        };
+        executeCommands(commands);
+        azureLogout();
+    }
+
+    private static void executeCommands(final String[] commands) throws IOException, InterruptedException {
+        if (commands != null && commands.length > 0) {
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java]: Only check `commands != null` is enough.

File: azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java [hunk 2]
```diff
@@ -9,26 +9,78 @@
...
+        azureLogin();
+        Runtime.getRuntime().exec(formatCommand(azResourceGroupDeleteTemplate,
+                new String[]{resourceGroupName, waitForOperationFinish ? "" : " --no-wait"})).waitFor();
+        azureLogout();
+    }
+
+    public static void createWindowsAzureWebApp(final String resourceGroupName, final String location,
+                                                final String servicePlanName, final String webAppName)
+            throws IOException, InterruptedException {
+        final String azResourceGroupCreateTemplate = "az group create -n %s --location %s";
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java]: `azResourceGroupCreateTemplate` looks like this string can be extracted as a const

File: azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java [hunk 3]
```diff
@@ -9,26 +9,78 @@
...
+        final String azAppServicePlanCreateTemplate = "az appservice plan create -g %s -n %s --sku S1";
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java]: Same as this one

File: azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java [hunk 4]
```diff
@@ -9,26 +9,78 @@
...
 public class TestUtils {
+    private static final String clientId = System.getenv("CLIENT_ID");
+    private static final String tenantId = System.getenv("TENANT_ID");
+    private static final String key = System.getenv("KEY");
+
+    private static final boolean isWindows = System.getProperty("os.name").contains("Windows");
+
+    private static void azureLogin() throws IOException, InterruptedException {
+        final String azLoginTemplate = "az login --service-principal -u %s -p %s --tenant %s";
+        Runtime.getRuntime().exec(formatCommand(azLoginTemplate, new String[]{clientId, key, tenantId})).waitFor();
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java]: We have a lot of `Runtime.getRuntime().exec(xxx).waitFor()` here. How about extract a method to wrap this?

Comment (SummerSun) [reply azure-webapp-maven-plugin/src/test/java/com/microsoft/azure/maven/webapp/utils/TestUtils.java]: good idea

**Pull request merged**